File: techgig/date_shopping/CandidateCode.cs
// Include namespace system
using System;
using System.IO;
public class CandidateCode
{
    public static void Main(String[] args)
    {
        int b = (int)Convert.ToInt64(Console.ReadLine());
        Console.ReadLine();
        
        int[] jeans =  Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
        
        int[] shirts =  Array.ConvertAll(Console.ReadLine().Split(), int.Parse);
        

        var k = 0;
        var mn = 0;
        var pre = -1;
        var flag = 0;
        foreach (int j in jeans)
        {
            foreach (int shirt in shirts)
            {
                var s = j + shirt;
                if (s <= b && flag == 0)
                {
                    pre = s;
                    flag = 1;
                }
                else if (s <= b)
                {
                    mn = s;
                    if (mn > pre)
                    {
                        pre = mn;
                    }
                }
                k = k + 1;
            }
        }
        Console.WriteLine(pre);
    }
}

File: techgig/date_shopping/CandidateCode.java
import java.util.Scanner;

class CandidateCode {
    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        int b = sc.nextInt();
        int[] jeans = new int[sc.nextInt()];
        int[] shirts = new int[sc.nextInt()];
        for (int i = 0; i < jeans.length; i++) {
            jeans[i] = sc.nextInt();
        }
        for (int i = 0; i < shirts.length; i++) {
            shirts[i] = sc.nextInt();
        }
        int k = 0;
        int mn = 0;
        int pre = -1;
        int flag = 0;
        for (int j : jeans) {
            for (int shirt : shirts) {
                int s = j + shirt;
                if (s <= b && flag == 0) {
                    pre = s;
                    flag = 1;
                } else if (s <= b) {
                    mn = s;
                    if (mn > pre) {
                        pre = mn;
                    }
                }
                k = k + 1;
            }
        }
        System.out.println(pre);
    }
}